cirq-core/cirq/ops/gate_operation.py

```diff
@@ -146,17 +146,18 @@ def _json_dict_(self) -> Dict[str, Any]:
     def _group_interchangeable_qubits(
         self,
     ) -> Tuple[Union['cirq.Qid', Tuple[int, FrozenSet['cirq.Qid']]], ...]:
-
         if not isinstance(self.gate, gate_features.InterchangeableQubitsGate):
             return self.qubits
-
-        groups: Dict[int, List['cirq.Qid']] = {}
-        for i, q in enumerate(self.qubits):
-            k = self.gate.qubit_index_to_equivalence_group_key(i)
-            if k not in groups:
-                groups[k] = []
-            groups[k].append(q)
-        return tuple(sorted((k, frozenset(v)) for k, v in groups.items()))
+        else:
+            return tuple(
+                (k, frozenset(g for _, g in kg))
+                for k, kg in itertools.groupby(
+                    enumerate(self.qubits),
+                    lambda i_q: cast(
+                        gate_features.InterchangeableQubitsGate, self.gate
+                    ).qubit_index_to_equivalence_group_key(i_q[0]),
```

Comment:
Could we break this up into a few simpler lines ? This is a little hard to read.

Reply:
Done.